flang/lib/Lower/OpenMP.cpp

```diff
@@ -54,6 +54,9 @@ getOmpObjectSymbol(const Fortran::parser::OmpObject &ompObject) {
                     Fortran::parser::Unwrap<Fortran::parser::ArrayElement>(
                         designator)) {
               sym = GetFirstName(arrayEle->base).symbol;
+            } else if (auto *structComp = Fortran::parser::Unwrap<
+                           Fortran::parser::StructureComponent>(designator)) {
+              sym = structComp->component.symbol;
```

Comment:
Is it OK to do the Structure Component portion in a separate patch?

Reply:
Can do! It was originally to emit an error for trying to offload allocatables in derived types as it's unsupported currently, but I didn't add it to the PR as it seems unnecessary, so I can drop this change and add it if it's ever required.